TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -398,18 +409,53 @@ void TTSSpeaker::createPipeline() {
     g_object_set(G_OBJECT(m_audioSink), "volume", (double) (m_defaultConfig.volume() / MAX_VOLUME), NULL);
 
     // Add elements to pipeline and link
+     if(m_pcmAudioEnabled) {
+        //Add raw audio caps
+       audiocaps = gst_caps_new_simple("audio/x-raw", "format", G_TYPE_STRING, "S16LE", "rate", G_TYPE_INT, 22050,
+                                "channels", G_TYPE_INT, 1, NULL);
+       if(audiocaps == NULL) {
+	   m_pcmAudioEnabled = false;
+           TTSLOG_ERROR("Unable to add audio caps for PCM audio.");
+           return ;
+       }
+      capsfilter = gst_element_factory_make ("capsfilter", NULL);
+      if (capsfilter) {
+         g_object_set (G_OBJECT (capsfilter), "caps", audiocaps, NULL);
+         gst_caps_unref(audiocaps);
+      }
+      else {
+          m_pcmAudioEnabled = false;
+          TTSLOG_ERROR( "Unable to create capsfilter for PCM audio.");
+          return;
+      }
+     }
     bool result = TRUE;
 #if defined(PLATFORM_BROADCOM)
+    if(!m_pcmAudioEnabled){
     gst_bin_add_many(GST_BIN(m_pipeline), m_source, decodebin, m_audioSink, NULL);
     result &= gst_element_link (m_source, decodebin);
     result &= gst_element_link (decodebin, m_audioSink);
+    }
+    else {
+        TTSLOG_INFO("PCM audio capsfilter  added to sink");
+        gst_bin_add_many(GST_BIN(m_pipeline), m_source, capsfilter, m_audioSink,NULL);
+        result = gst_element_link_many (m_source,capsfilter,m_audioSink,NULL);
+    }
 #elif defined(PLATFORM_AMLOGIC)
+    if(!m_pcmAudioEnabled){
     gst_bin_add_many(GST_BIN(m_pipeline), m_source, parser, decodebin, convert, resample, m_audioSink, NULL);
     result &= gst_element_link (m_source, parser);
     result &= gst_element_link (parser, decodebin);
     result &= gst_element_link (decodebin, convert);
     result &= gst_element_link (convert, resample);
     result &= gst_element_link (resample, m_audioSink);
+    }
+    else {
+         //TODO find PCM playback on Amlogic
```

Comment:
same applies for AMLOGIC as well, i.e though the m_pcmAudioEnabled is set, the parse, decodebin elements are being created and left unattended